src/serializer/functions.js

```diff
@@ -295,11 +298,12 @@ export class Functions {
       invariant(e1 !== undefined);
       if (e1.result instanceof Completion && !e1.result instanceof PossiblyNormalCompletion) {
         let error = new CompilerDiagnostic(
-          `Additional function ${fun1Name} may terminate abruptly`,
+          `Additional function ${fun1Name} will terminate abruptly`,
           e1.result.location,
           "PP1002",
-          "FatalError"
+          "RecoverableError"
         );
+        // We generate correct code in this case, but the user probably doesn't want us to emit an unconditional throw
```

Comment:
Since this is marked as recoverable (and is recoverable) we have to honor the result of handleError.